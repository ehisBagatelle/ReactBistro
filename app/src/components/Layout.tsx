import React, { ReactNode } from "react";
import { Text, View, Image, ScrollView } from "react-native";
import { StyleSheet } from "react-native";


interface Props {
  children?: ReactNode;
}

const Layout: React.FC<Props> = ({ children }) => {
  return (
    <View style={styles.container}>
      <View style={styles.header}>
        <Image source={require("../../assets/icons8-bistro-64.png")} />
        <Text style={styles.logoText}>React Bistro</Text>
      </View>
      <View style={{ width: "100%" }}>{children}</View>
    </View>
  );
};

const styles = StyleSheet.create({
  container: {
    flex: 1,
    backgroundColor: "#fff",
    alignItems: "center",
    justifyContent: "flex-start",
    width: '100%'
  },
  header:{
    display:'flex',
    flexDirection: 'row',
    alignItems: 'center',
    justifyContent:'flex-start',
    width: '100%',
    paddingVertical: 10
  },
  logoText:{
    color: "#000",
    fontSize: 60,
    fontStyle: "normal",
    fontWeight: "400",
  }
});

export default Layout;
