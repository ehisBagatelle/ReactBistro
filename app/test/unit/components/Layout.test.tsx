import React from 'react';
import {Text} from 'react-native'
import renderer from 'react-test-renderer';
import Layout from '../../../src/components/Layout';

describe('Layout component', () => {
  it('renders correctly with children', () => {
    const mockChildren = <Text>Mock Child</Text>;
    const tree = renderer.create(<Layout>{mockChildren}</Layout>).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders correctly without children', () => {
    const tree = renderer.create(<Layout />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
