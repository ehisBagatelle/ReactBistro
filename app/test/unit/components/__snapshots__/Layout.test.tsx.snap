// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Layout component renders correctly with children 1`] = `
<View
  style={
    {
      "alignItems": "center",
      "backgroundColor": "#fff",
      "flex": 1,
      "justifyContent": "flex-start",
      "width": "100%",
    }
  }
>
  <View
    style={
      {
        "alignItems": "center",
        "display": "flex",
        "flexDirection": "row",
        "justifyContent": "flex-start",
        "paddingVertical": 10,
        "width": "100%",
      }
    }
  >
    <Image
      source={
        {
          "testUri": "../../../assets/icons8-bistro-64.png",
        }
      }
    />
    <Text
      style={
        {
          "color": "#000",
          "fontSize": 60,
          "fontStyle": "normal",
          "fontWeight": "400",
        }
      }
    >
      React Bistro
    </Text>
  </View>
  <View
    style={
      {
        "width": "100%",
      }
    }
  >
    <Text>
      Mock Child
    </Text>
  </View>
</View>
`;

exports[`Layout component renders correctly without children 1`] = `
<View
  style={
    {
      "alignItems": "center",
      "backgroundColor": "#fff",
      "flex": 1,
      "justifyContent": "flex-start",
      "width": "100%",
    }
  }
>
  <View
    style={
      {
        "alignItems": "center",
        "display": "flex",
        "flexDirection": "row",
        "justifyContent": "flex-start",
        "paddingVertical": 10,
        "width": "100%",
      }
    }
  >
    <Image
      source={
        {
          "testUri": "../../../assets/icons8-bistro-64.png",
        }
      }
    />
    <Text
      style={
        {
          "color": "#000",
          "fontSize": 60,
          "fontStyle": "normal",
          "fontWeight": "400",
        }
      }
    >
      React Bistro
    </Text>
  </View>
  <View
    style={
      {
        "width": "100%",
      }
    }
  />
</View>
`;
